tool/release/version/Bump.kt

```diff
@@ -35,34 +27,44 @@ fun bumpVersion(version: String): String {
                     ).toString()
             lastVersionComponent = versionSubComponents.joinToString("-")
         } catch (b: NumberFormatException) {
-            throw RuntimeException("invalid version: ${version}")
+            throw RuntimeException("invalid version: $version")
         }
     }
     versionComponents[versionComponents.lastIndex] = lastVersionComponent
     return versionComponents.joinToString(".")
 }
 
+fun gitCommitAndPush(workspacePath: Path, newVersion: String) {
+    shell.execute(listOf("git", "add", "VERSION"), baseDir = workspacePath)
+    shell.execute(listOf("git", "commit", "-m", "\"Bump version number to $newVersion\""), baseDir = workspacePath)
+    val maxRetries = 3
+    var retryCount = 0
+    while (retryCount < maxRetries) {
+        shell.execute(listOf("git", "pull"), baseDir = workspacePath)
+        try {
+            shell.execute(listOf("git", "push"), baseDir = workspacePath)
+            break
+        } catch (e: Exception) {
+            // cover the unlikely but possible edge case where someone else has already pushed
+            retryCount++
+            if (retryCount == maxRetries) throw e
+        }
+    }
+}
+
 fun main() {
     val workspaceDirectory = System.getenv("BUILD_WORKSPACE_DIRECTORY")
             ?: throw RuntimeException("Not running from within Bazel workspace")
-    val githubToken = System.getenv("GITHUB_TOKEN")
-            ?: throw RuntimeException("GITHUB_TOKEN environment variable is not set")
-    val repoOwner = System.getenv("GRABL_OWNER")
-            ?: throw RuntimeException("GRABL_OWNER environment variable is not set")
-    val repoName = System.getenv("GRABL_REPO")
-            ?: throw RuntimeException("GRABL_REPO environment variable is not set")
+    val workspacePath = Paths.get(workspaceDirectory)
+
+    shell.execute(listOf("git", "checkout", "master"), baseDir = workspacePath)
 
-    val versionFile = Paths.get(workspaceDirectory, "VERSION")
+    val versionFile = workspacePath.resolve("VERSION")
```

Comment:
While we can assume most repos will have a `VERSION` file, it's still nice to do an assertion and throw a meaningful error message when it doesn't:

```
val versionFile = workspacePath.resolve("VERSION")
if (!versionFile.exists()) throw RuntimeException("...")
```